features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/DataQueueHandler.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.async
+
+import com.datadog.android.sessionreplay.internal.recorder.SystemInformation
+import com.datadog.android.sessionreplay.model.MobileSegment
+
+internal interface DataQueueHandler {
+    fun addResourceItem(
```

Comment:
Resolving the bitmap is on the mainthread, as we did previously. However, adding the resource item to the queue should be in the background since serializeBitmap is on a background thread